runtime/ts/storage/pouchdb/pouch-db-collection.ts

```diff
@@ -0,0 +1,364 @@
+      };
+
+      return await Promise.all(items.map(retrieveItem));
+    }
+    return (await this.getModel()).toLiteral();
+  }
+
+  async toList() {
+    return (await this._toList()).map(item => item.value);
+  }
+
+  /**
+   * Returns an array of values for each of the specified ids.
+   *
+   * @param ids items to fetch from the underlying CRDT model.
+   * @return an array of values from the underlying CRDT
+   */
+  async getMultiple(ids: string[]) {
+    assert(!this.referenceMode, "getMultiple not implemented for referenceMode stores");
+    const model = await this.getModel();
+    return ids.map(id => model.getValue(id));
+  }
+
+  /**
+   * Store multiple values with the given keys in the Collection.
+   * TODO(lindner): document originatorId, which is unused.
+   */
+  async storeMultiple(values, keys, originatorId=null) {
+    assert(!this.referenceMode, "storeMultiple not implemented for referenceMode stores");
+
+    this.getModelAndUpdate((crdtmodel) => {
+      values.map(value => crdtmodel.add(value.id, value, keys));
+      return crdtmodel;
+    });
+  }
+
+
+  /**
+   * Get a specific Id value previously set using store(). 
+   *
+   * @remarks Note that the id referred to here is not the same as
+   * used in the constructor.
+   */
+  async get(id) {
+    if (this.referenceMode) {
+      const ref = (await this.getModel()).getValue(id);
+      if (ref == null) {
+        return null;
+      }
+      await this.ensureBackingStore();
+      const result = await this.backingStore.get(ref.id);
+      return result;
+    }
+    
+    const model = await this.getModel();
+    return model.getValue(id);
+  }
+
+  /**
+   * Store the specific value to the collection.  Value must include an id entry.
+   *
+   * @param value A data object with an id entry that is used as a key.
+   * @param keys The CRDT keys used to store this object
+   * @param orginatorId TBD passed to event listeners
+   */
+  async store(value, keys: string[], originatorId=null) : Promise<void> {
+    assert(keys != null && keys.length > 0, 'keys required');
+    const id = value.id;
+    
+    const changeEvent = {value, keys, effective: undefined};
+
+    if (this.referenceMode) {
+      const referredType = this.type.primitiveType();
+      const storageKey = this.storageEngine.baseStorageKey(referredType, this.storageKey);
+
+      // Update the referred data
+      await this.getModelAndUpdate((crdtmodel) => {
+        changeEvent.effective = crdtmodel.add(value.id, {id: value.id, storageKey}, keys);
+        return crdtmodel;
+      });
+
+      await this.ensureBackingStore();
+      await this.backingStore.store(value, keys);
+    } else {
+      await this.getModelAndUpdate((crdtmodel) => {
+        // check for existing keys?
+        changeEvent.effective = crdtmodel.add(value.id, value, keys);
+        return crdtmodel;
+      });
+    }
+
+    this.version++;
+
+    // 2. Notify Listeners
```

Comment:
Where is `1.`? :) Maybe it's 180 and lost or never had the prefix?

Reply:
this was cribbed from another storage driver.  removed the 2.